ocaml/typing/jkind_intf.ml

```diff
@@ -240,7 +240,9 @@ module History = struct
     | Primitive of Ident.t
     | Immediate_polymorphic_variant
 
-  type immediate64_creation_reason = Separability_check
+  type immediate64_creation_reason =
+    | Separability_check
+    | Primitive of Ident.t
```

Comment:
Why do we need this? There aren't any primitive `immediate64`s, I think.

Reply:
Yes, previously I thought `int32` is `immediate64` which caused bugs. I've fixed it but didn't remove  this. I will remove it.